### DISCUSSION TITLE
Como traerme un atributo generado por el ORM que no existia en ninguna de las clases originales

### DISCUSSION THREAD

**Discussion by julchat** [dds-jv/foro#163] [Consultas de persistencia] [answered]
Hola, sé que es un poco tarde para preguntarlo, pero si alguno sabe la respuesta me ayudaría mucho.

Cuestión, en la entrega pasada usamos annotations como JoinColumn(<nombre columna>) en relaciones One to Many y usamos para las entidades mapeo de herencia JOINED TABLE, en donde se genera un ID PK FK en las subclases que referencia a la clase madre dentro de las tablas.

Ahora, para nosotros traernos algo de la database, debemos usar createQuery. El problema es que por lo que vi en createQuery vos tenes que tratar como tus tablas a las clases de tu dominio, no a las que se crearon en la database por el ORM. Entonces, atributos que están en la parte de persistencia no puedo accederlos por el dominio (porque no son atributos de ninguna clase) y por lo tanto no puedo presentarlos.

Si alguno conoce alguna solucion alternativa o si estoy haciendo mal las cosas porfa me avisa.
Gracias,
Saludos

**Comment by julchat** [comment]
```
TypedQuery<infoBase> queryTodoBases = entityManager.createQuery("SELECT b.nombreFicticio, b.descripcion,"
				+ " c.nombre, e.id, e.id_organizacion, b.id_juridica FROM Base b INNER JOIN b.id_entidad_madre as e INNER JOIN e.categoria_id as c"
				, infoBase.class);

```

esa es la consulta que quiero hacer,

**Comment by gastonprieto** [comment]
Buenas @julchat !

No estoy muy seguro como tenes el modelo de objetos. Pero tu razonamiento es correcto no deberías acceder a los campos de la tabla.

Si nos contas como es tu modelo de objetos y que queres hacer semanticamente seguro te podemos ayudar un poco mas.

**Comment by julchat** [comment, marked as answer]
Está, no me había dado cuenta que las clases estaban heredando de persistent entity y por eso no vi el ID. De eso ya me quedo claro
Gracias!
